packages/flutter/lib/src/widgets/tap_region.dart

```diff
@@ -403,12 +407,14 @@ class TapRegion extends SingleChildRenderObjectWidget {
 
   @override
   RenderObject createRenderObject(BuildContext context) {
+    final ModalRoute<Object?>? route = ModalRoute.of(context);
+
     return RenderTapRegion(
       registry: TapRegionRegistry.maybeOf(context),
       enabled: enabled,
       consumeOutsideTaps: consumeOutsideTaps,
       behavior: behavior,
-      onTapOutside: onTapOutside,
+      onTapOutside: (route?.isCurrent ?? true) ? onTapOutside : null,
```

Comment:
Can we use https://api.flutter.dev/flutter/widgets/ModalRoute/isCurrentOf.html here? That uses an aspect of the modal route, which would be better for rebuilds.

Reply:
Yes it works, thanks.

Reply:
Didn't realize ModalRoute was an inherited model, good to know!